content/en/docs/concepts/security/hardening-guide/authentication-mechanisms.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Hardening Guide - Authentication Mechanisms
+description: >
+  Information on authentication options in Kubernetes and their security properties.
+content_type: concept
+weight: 90
+---
+
+<!-- overview -->
+
+Selecting the appropriate authentication mechanism(s) is a crucial aspect of securing your cluster.
+Kubernetes provides several built-in mechanisms, each with its own strengths and weaknesses that 
+should be carefully considered when choosing the best authentication mechanism for your cluster.
+
+In general, it is recommended to enable as few authentication mechanisms as possible to simplify 
+user management and prevent cases where users retain access to a cluster that is no longer required.
+
+It is important to note that Kubernetes does not have an in-built user database within the cluster. 
+Instead, it takes user information from the configured authentication system and uses that to make 
+authorization decisions. Therefore, to audit user access, you need to review credentials from every 
+configured authentication source.
+
+For production clusters with multiple users directly accessing the Kubernetes API, it is 
+recommended to use external authentication sources such as OIDC. The internal authentication 
+mechanisms, such as client certificates and service account tokens, described below, are not 
+suitable for this use-case.
+
+<!-- body -->
+
+## X.509 client certificate authentication {#x509-client-certificate-authentication}
+
+Kubernetes leverages [X.509 client certificate](/docs/reference/access-authn-authz/authentication/#x509-client-certs) 
+authentication for system components, such as when the Kubelet authenticates to the API Server. 
+While this mechanism can also be used for user authentication, it might not be suitable for 
+production use due to several restrictions:
+
+- Client certificates cannot be individually revoked. Once compromised, a certificate can be used 
+by an attacker until it expires. To mitigate this risk, it is recommended to configure short 
+lifetimes for user authentication credentials created using client certificates.
```

Comment:
The current indentation works, but some more indentation may help improve readability for future revisions. For example:

```suggestion
- Client certificates cannot be individually revoked. Once compromised, a certificate can be used 
  by an attacker until it expires. To mitigate this risk, it is recommended to configure short 
  lifetimes for user authentication credentials created using client certificates.
```